massa-grpc/src/stream/new_slot_execution_outputs.rs

```diff
@@ -400,14 +439,170 @@ fn filter_map_exec_output(
     if let Some(execution_event_filter) = &filters.execution_event_filter {
         if execution_event_filter.none.is_some() {
             exec_output.events.clear();
+        } else {
+            exec_output.events.0.retain(|event| {
+                if let Some(is_failure) = execution_event_filter.is_failure {
+                    if event.context.is_error != is_failure {
+                        return false;
+                    }
+                }
+
+                if let (Some(original_operation_ids), Some(origin_operation_id)) = (
+                    execution_event_filter.original_operation_ids.clone(),
+                    event.context.origin_operation_id,
+                ) {
+                    if !original_operation_ids.contains(&origin_operation_id) {
+                        return false;
+                    }
+                }
+
+                if let Some(caller_addresses) = execution_event_filter.caller_addresses.clone() {
+                    if !caller_addresses
+                        .into_iter()
+                        .any(|caller_address| event.context.call_stack.contains(&caller_address))
+                    {
+                        return false;
+                    };
+                }
+                //TODO to be confirmed
+                if let Some(emitter_addresses) = execution_event_filter.emitter_addresses.clone() {
+                    if !emitter_addresses
+                        .into_iter()
+                        .any(|emitter_address| event.context.call_stack.contains(&emitter_address))
+                    {
+                        return false;
+                    };
+                }
+
+                true
+            });
+
+            if exec_output.events.0.is_empty() {
+                return None;
+            }
         }
     }
 
     if let Some(async_pool_changes_filter) = &filters.async_pool_changes_filter {
         if async_pool_changes_filter.none.is_some() {
             exec_output.state_changes.async_pool_changes.0.clear();
+        } else {
+            exec_output
+                .state_changes
+                .async_pool_changes
+                .0
+                .retain(|_, change| {
+                    match change {
+                        SetUpdateOrDelete::Set(value) => {
+                            if let Some(change_types) = &async_pool_changes_filter.change_types {
+                                if !change_types
+                                    .contains(&(grpc_model::AsyncPoolChangeType::Set as i32))
+                                {
+                                    return false;
+                                }
+                            }
+                            //TODO to be confirmed
+                            if let Some(handlers) = &async_pool_changes_filter.handlers {
```

Comment:
@AurelienFT same here about `handler`

Reply:
I have renamed `handler` to `function` because it was his real usage. So yes `handler` before == `function` now.